tests/Application/src/Migrations/Version1.php

```diff
@@ -4,185 +4,198 @@
 
 namespace DoctrineMigrations;
 
-use Doctrine\Migrations\AbstractMigration;
 use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
 
 /**
  * Auto-generated Migration: Please modify to your needs!
  */
-class Version20170912085504 extends AbstractMigration
+final class Version1 extends AbstractMigration
```

Comment:
If this migration is used only in test env I'm ok with this change. Nonetheless, why can't we leave the previous implementation and add new migrations on top of them?

Also, we cannot move to `schema:upgrade --force` as test migrations are compound with main migrations and we need to run all of them to test if they work. 

Also, we are working with a new approach to migration: https://github.com/Sylius/Sylius/pull/11389 Hopefully it will reduce such problems in the future.

Reply:
The problem is that this migration contains all the sylius database changes, which will often change. Actually I'm in favor of a schema:update instead of using migrations for that.

Reply:
I agree that for me the best way to install a plugin is to run the `doctrine:migration:diff` command, for Migrations.

Reply:
After re-reading the thread I think we're all talking of different things.

First, @jacquesbh this migration is only for the behat test environment. It's not related to the migrations of the plugin. This is another debate.

Second, I talk about `do:schema:update` for this behat application only. I understand running the plugin migration is an actual test. And I also understand that since the plugin is running at all times the `do:schema:up` will include db changes of the plugin. But my point is that we could enable dynamically the plugin while building, and after the generation of the database.


Finally, I forgot to answer clearly to this message:

> Nonetheless, why can't we leave the previous implementation and add new migrations on top of them?

One of the issues is that the migration is not up to date inside the app. I can ofc just add a new one but I think it's far more clear to have only 1 migration. This will hopefully avoid @jacquesbh [to misunderstand](https://github.com/Sylius/RefundPlugin/pull/211#issuecomment-634695467) the thing for example ;).

Reply:
According to the one migration, we used to copy additional, required migration from the UPGRADE.md file. You can see it here: https://github.com/Sylius/InvoicingPlugin/commit/f75f7763709d3a5fd1acc77d2f5a65114d49a1a8. The reasoning behind it is that at the end tests application - is just Sylius application and we also test upgrade files this way. What is more, we are keeping them in sync, which reduce cost of maintenance.